nomenclature/codelist.py

```diff
@@ -201,3 +203,36 @@ def from_directory(
 
         # iterate over the list to guard against silent replacement of duplicates
         return CodeList(name=name, mapping=code_list)
+
+    def to_excel(self, excel_writer, sheet_name="definitions"):
```

Comment:
Is there a specific reason that the user would want to provide their own `excel_writer`?
If not I would find it more convenient that the argument would only contain the path to the target file and be renamed `target`, `output_file_path` or something along those lines.

Reply:
If you look at my analysis scripts for the IPCC SR15 (see last cells [here](https://data.ene.iiasa.ac.at/sr15_scenario_analysis/assessment/sr15_2.0_categories_indicators.html)), I wanted to write several sheets within one xlsx file. For that, using the **ExcelWriter** is quite helpful.

Reply:
Ah alright, that is a fair use case. This could also be useful in case we ever want to implement `to_excel` on a `DataStructureDefinition` basis where we would export and entire `DataStructureDefinition` in a single excel file into different sheets.